uint/src/uint.rs

```diff
@@ -321,15 +368,9 @@ macro_rules! impl_mulassign_from {
 
 #[inline(always)]
 #[doc(hidden)]
-pub fn mul_u32(a: (u64, u64), b: u64, carry: u64) -> (u64, u64) {
-	let upper = b * a.0;
-	let lower = b * a.1;
-
-	let (res1, overflow1) = lower.overflowing_add(upper << 32);
-	let (res2, overflow2) = res1.overflowing_add(carry);
-
-	let carry = (upper >> 32) + overflow1 as u64 + overflow2 as u64;
-	(res2, carry)
+pub fn mul_u64(a: u64, b: u64, carry: u64) -> (u64, u64) {
```

Comment:
Does it need to be public?

Reply:
Yes, this function is defined in uint crate, but the uint type itself is instantiated in the module it is defined (via macro).

Reply:
re-thinking about this. Not sure I follow the reasoning. This is a free function, a utility, called only from `overflowing_mul_u64` I think. It is always inlined, so why can't it be a part of the `impl $name` for each type? (and be private)? Sorry if dumb q!

Reply:
I guess the same q applies to the `split_128` as well?

Reply:
TBH, I don't know the reasoning that was used before, maybe it will affect generated binary size somehow, but it seems like a good idea to me. :+1: